src/Core/Configurations/ConfigFileWatcher.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using Azure.DataApiBuilder.Config;
+using Path = System.IO.Path;
+
+namespace Azure.DataApiBuilder.Core.Configurations;
+
+public class ConfigFileWatcher
+{
+    private FileSystemWatcher? _fileWatcher;
+    private readonly RuntimeConfigProvider? _configProvider;
+
+    public ConfigFileWatcher(RuntimeConfigProvider configProvider)
+    {
+        FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)configProvider.ConfigLoader;
+        string configFilePath = loader.ConfigFilePath;
+        IFileSystem fileSystem = (IFileSystem)loader.FileSystem;
+        string? directoryName = Path.GetDirectoryName(configFilePath);
+        directoryName = string.IsNullOrWhiteSpace(directoryName) ?
+            fileSystem.Directory.GetCurrentDirectory() :
+            directoryName;
```

Comment:
We want that `path` to come from the FileSystem here. And when we invoke the file watcher's constructor, we only have access to the loader since we are inside of the `RuntimeConfigProvider`. How are you proposing we get the `path` that we need here if not from the `FileSystem` contained in the loader?

Reply:
do you mean adding something like a `GetPath()` function to the Loader that we can call to have to internally get the path for us without making the `FileSystem` public? We could call this from the `RuntimeConfigProvider` before we instantiate the file watcher.